presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -205,6 +207,23 @@ private static ConnectorPageSource createSelectivePageSource(
         throw new IllegalStateException("Could not find a file reader for split " + split);
     }
 
+    public static List<HiveColumnHandle> renumberPartitioningColumns(List<HiveColumnHandle> columns)
```

Comment:
- make this method private
- `renumber` is a bit too generic; can we find a more specific name? for example, assignUniqueIndicesToPartitionColumns
- this method is only for partitioning keys, hence, nextPrefilledColumnIndex name is confusing; let's change to nextIndex
- nit: I'd rename renumberedColumns to newColumns
- use column.isPartitionKey() instead of column.getHiveColumnIndex() == -1

```
    private static List<HiveColumnHandle> assignUniqueIndicesToPartitionColumns(List<HiveColumnHandle> columns)
    {
        // Gives a distinct hiveColumnIndex to partitioning columns. Columns are identified by these indices in the rest of the
        // selective read path.
        int nextIndex = MAX_PARTITION_KEY_COLUMN_INDEX;
        ImmutableList.Builder<HiveColumnHandle> newColumns = ImmutableList.builder();
        for (HiveColumnHandle column : columns) {
            if (column.isPartitionKey()) {
                newColumns.add(new HiveColumnHandle(column.getName(), column.getHiveType(), column.getTypeSignature(), nextIndex, column.getColumnType(), column.getComment(), column.getRequiredSubfields()));
                nextIndex--;
            }
            else {
                newColumns.add(column);
            }
        }
        return newColumns.build();
    }
```